### ISSUE TITLE
Installation / setup fails when compiling something, something ClassyPrelude

### ISSUE THREAD

**Issue by alxlo** [nek0/yammat#117]
I tried to follow installation instructions but the build fails in a late stage with
```
Preprocessing library classy-prelude-1.4.0...
[1 of 2] Compiling Paths_classy_prelude ( dist/build/autogen/Paths_classy_prelude.hs, dist/build/Paths_classy_prelude.o )
[2 of 2] Compiling ClassyPrelude    ( src/ClassyPrelude.hs, dist/build/ClassyPrelude.o )

src/ClassyPrelude.hs:438:13: error:
    Not in scope: ‘STM.always’
    Module ‘Control.Concurrent.STM’ does not export ‘always’.

src/ClassyPrelude.hs:443:21: error:
    Not in scope: ‘STM.alwaysSucceeds’
    Module ‘Control.Concurrent.STM’ does not export ‘alwaysSucceeds’.
cabal: Leaving directory '/tmp/cabal-tmp-8/classy-prelude-1.4.0'
cabal: Error: some packages failed to install:
JuicyPixels-scale-dct-0.1.2 depends on fft-0.1.8.6 which failed to install.
classy-prelude-1.4.0 failed during the building phase. The exception was:
ExitFailure 1
classy-prelude-conduit-1.4.0 depends on classy-prelude-1.4.0 which failed to
install.
classy-prelude-yesod-1.4.0 depends on classy-prelude-1.4.0 which failed to
install.
fft-0.1.8.6 failed during the configure step. The exception was:
ExitFailure 1
```
To reproduce my installation enverionment and procedure, here is the Dockerfile that has been used:

```Dockerfile
FROM ubuntu:18.04
MAINTAINER Alexander Lorz (bigalex@c3d2.de)

ENV DEBIAN_FRONTEND=noninteractive
ENV LANGUAGE de_DE.UTF-8
ENV LANG de_DE.UTF-8
ENV TZ "Europe/Berlin"


# install patches, locale settings
# Timezone settings
# https://stackoverflow.com/questions/40234847/docker-timezone-in-ubuntu-16-04-image

RUN apt-get update \
  && apt-get --no-install-recommends -y dist-upgrade \
  && apt-get --no-install-recommends install -y locales \
  && locale-gen de_DE.UTF-8 \
  && dpkg-reconfigure --frontend noninteractive locales \
  && apt-get -qqy --no-install-recommends install \
    language-pack-de \
  && apt-get --no-install-recommends install -y tzdata \
  && echo "${TZ}" > /etc/timezone \
  && rm /etc/localtime \
  && dpkg-reconfigure --frontend noninteractive tzdata

# install haskell and cabal

RUN apt-get --no-install-recommends -y install \
    ghc cabal-install zlibc zlib1g-dev

RUN rm -rf /root/.cabal

RUN  cabal update 
# RUN cabal install cabal-install 
RUN apt-get --no-install-recommends -y install \
    build-essential alex happy libpq-dev postgresql \
    libfftw3-dev 
RUN apt-get --no-install-recommends -y install git ca-certificates
RUN git clone https://github.com/nek0/yammat.git 
  
RUN cd yammat \
  && cabal install --only-dependencies

RUN cabal build
```

**Comment by astro** [comment]
Please give compiler version: `ghc -V`

**Comment by alxlo** [comment]
```
Glasgow Haskell Compiler, Version 8.0.2, stage 2 booted by GHC version 8.0.2
Using binary package database: /usr/lib/ghc/package.conf.d/package.cache
loading package database /usr/lib/ghc/package.conf.d
wired-in package ghc-prim mapped to ghc-prim-0.5.0.0
wired-in package integer-gmp mapped to integer-gmp-1.0.0.1
wired-in package base mapped to base-4.9.1.0
wired-in package rts mapped to rts
wired-in package template-haskell mapped to template-haskell-2.11.1.0
wired-in package ghc mapped to ghc-8.0.2
wired-in package dph-seq not found.
wired-in package dph-par not found.
Hsc static flags: 
*** Deleting temp files:
Deleting: 
*** Deleting temp dirs:
Deleting: 
ghc: no input files
Usage: For basic information, try the `--help' option.
```

**Comment by astro** [comment]
```diff
diff --git yammat.cabal yammat.cabal
index e08bf1d..98b42c1 100644
--- yammat.cabal
+++ yammat.cabal
@@ -72,6 +72,7 @@ library
                  , yesod-static                  >= 1.4.0.3
                  , yesod-form                    >= 1.4.0
                  , classy-prelude                >= 0.10.2
+                 , stm                            < 2.5
                  , classy-prelude-conduit        >= 0.10.2
                  , classy-prelude-yesod          >= 0.10.2
                  , bytestring                    >= 0.9
```
